sample-apps/deepedit/main.py

```diff
@@ -37,37 +39,39 @@
 
 class MyApp(MONAILabelApp):
     def __init__(self, app_dir, studies, conf):
-        self.network = DynUNetV1(
-            spatial_dims=3,
-            in_channels=3,
-            out_channels=1,
-            kernel_size=[
+        network_params = {
+            "spatial_dims": 3,
+            "in_channels": 3,
+            "out_channels": 1,
+            "kernel_size": [
                 [3, 3, 3],
                 [3, 3, 3],
                 [3, 3, 3],
                 [3, 3, 3],
                 [3, 3, 3],
                 [3, 3, 3],
             ],
-            strides=[
+            "strides": [
                 [1, 1, 1],
                 [2, 2, 2],
                 [2, 2, 2],
                 [2, 2, 2],
                 [2, 2, 2],
                 [2, 2, 1],
             ],
-            upsample_kernel_size=[
+            "upsample_kernel_size": [
                 [2, 2, 2],
                 [2, 2, 2],
                 [2, 2, 2],
                 [2, 2, 2],
                 [2, 2, 1],
             ],
-            norm_name="instance",
-            deep_supervision=False,
-            res_block=True,
-        )
+            "norm_name": "instance",
+            "deep_supervision": False,
+            "res_block": True,
+        }
+        self.network = DynUNetV1(**network_params)
+        self.network_with_dropout = DynUNetV1(**network_params, dropout=0.2)
```

Comment:
Are both of these trained independently? how do you know they are learning similar things? shouldn't the weights be shared so that they converge with and without dropout?